src/models/Link.php

```diff
@@ -517,8 +517,10 @@ public function getTypes()
         if ($this->allowed_types) {
             // Prioritise local field over global settings
             $allowed_types = $this->allowed_types;
+        } else {
+            $allowed_types = $this->config()->get('allowed_types');
         }
-        if ($allowed_types = $this->config()->get('allowed_types')) {
+        if ($allowed_types) {
```

Comment:
There's a potential scope issue here where $allowed_types is evaluated before it is set.  Try this instead:
```php
            ...
        $allowed_types = $this->config()->get('allowed_types');
        if ($this->allowed_types) {
            // Prioritise local field over global settings
            $allowed_types = $this->allowed_types;
        }
        if ($allowed_types) {
            ...
```

Reply:
That's right. I resolved the issue.